template.yaml

```diff
@@ -1,67 +1,87 @@
+  Needs to be periodically re-triggered because the IP address change.
+  Data transfers within the same region between AWS resources are free.
+  Restricts downloading files from this bucket to only AWS resources (e.g. EC2 , Lambda) within the same region as this bucket.
+  This will not allow even the owner of the bucket to download objects in this bucket when not using an AWS resource in the same region!
 Metadata:
   AWS::ServerlessRepo::Application:
-    Name: "lambda-template"
-    Description: "A GH template for quickly starting a new AWS lambda project."
+    Name: "cfn-cr-same-region-bucket-download"
+    Description: "Add restriction on S3 bucket to only allow download from AWS resources in the same region. Needs to be periodically re-triggered because the IP address change."
     Author: "Sage-Bionetworks"
     SpdxLicenseId: "Apache-2.0"
     # paths are relative to .aws-sam/build directory
     LicenseUrl: "LICENSE"
     ReadmeUrl: "README.md"
     Labels: ["serverless", "template", "github", "quick-start"]
-    HomePageUrl: "https://github.com/Sage-Bionetworks-IT/lambda-template"
-    SemanticVersion: "0.0.3"
-    SourceCodeUrl: "https://github.com/Sage-Bionetworks-IT/lambda-template/tree/0.0.3"
-
-# More info about Globals: https://github.com/awslabs/serverless-application-model/blob/master/docs/globals.rst
-Globals:
-  Function:
-    Timeout: 3
+    HomePageUrl: "https://github.com/Sage-Bionetworks-IT/cfn-cr-same-region-bucket-download"
+    SemanticVersion: "0.0.1"
+    SourceCodeUrl: "https://github.com/Sage-Bionetworks-IT/cfn-cr-same-region-bucket-download/tree/0.0.1"
 
 Resources:
-  HelloWorldFunction:
-    Type: AWS::Serverless::Function # More info about Function Resource: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction
-    Properties:
-      CodeUri: .
-      Handler: hello_world/app.lambda_handler
-      Runtime: python3.7
-      Role: !GetAtt FunctionRole.Arn
-      Events:
-        HelloWorld:
-          Type: Api # More info about API Event Source: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#api
-          Properties:
-            Path: /hello
-            Method: get
-
-  FunctionRole:   # execute lambda function with this role
-    Type: AWS::IAM::Role
+  RestrictBucketDownloadRegionRole:
+    Type: "AWS::IAM::Role"
     Properties:
       AssumeRolePolicyDocument:
-        Version: 2012-10-17
+        Version: "2012-10-17"
         Statement:
           - Effect: Allow
             Principal:
-              Service:
-                - lambda.amazonaws.com
-            Action:
-              - sts:AssumeRole
+              Service: lambda.amazonaws.com
+            Action: sts:AssumeRole
+      Path: "/"
       ManagedPolicyArns:
         - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+      Policies:
```

Comment:
this is an inline policy.  it's a best practice not use inline policies but rather create a new managed policy and add it to the  `ManagedPolicyArns` list.

Reply:
is that a policy I should create in this file or somewhere else?